rusk-wallet/src/main.rs

```diff
@@ -247,64 +242,67 @@ async fn main() -> Result<(), Error> {
 async fn exec() -> Result<(), Error> {
     use CliCommand::*;
 
-    // parse cli arguments
-    let cfg: WalletCfg = WalletCfg::parse();
+    // load configuration (or use default)
+    let mut cfg = Config::load().unwrap_or_else(Config::default);
+
+    // parse user args and merge with static config
+    let args = WalletArgs::parse();
+    let cmd = args.command.clone();
+    cfg.merge(args);
 
     // create directories
-    let data_dir = cfg.data_dir.unwrap_or_else(LocalStore::default_data_dir);
-    LocalStore::create_dir(&data_dir)?;
+    LocalStore::create_dir(&cfg.wallet.data_dir)?;
 
     // get command or default to interactive mode
-    let cmd = cfg.command.unwrap_or(CliCommand::Interactive);
+    let cmd = cmd.unwrap_or(CliCommand::Interactive);
 
     // request auth for wallet (if required)
-    let pwd = if cmd.uses_wallet() || cfg.wallet_file.is_some() {
+    let pwd = if cmd.uses_wallet() || cfg.wallet.file.is_some() {
         prompt::request_auth("Please enter wallet password")
     } else {
         blake3::hash("".as_bytes())
     };
 
     // prepare wallet path
     let mut path_override = false;
-    let wallet_path = if let Some(p) = cfg.wallet_file {
+    let wallet_path = if let Some(ref p) = cfg.wallet.file {
         path_override = true;
         p.with_extension("dat")
     } else {
-        let wallet_name = cfg
-            .wallet_name
-            .unwrap_or_else(LocalStore::default_wallet_name);
         let mut pb = PathBuf::new();
-        pb.push(&data_dir);
-        pb.push(&wallet_name);
+        pb.push(&cfg.wallet.data_dir);
+        pb.push(&cfg.wallet.name);
         pb.set_extension("dat");
         pb
     };
 
     // start our local store
     let store = match cmd {
-        Create => create(wallet_path, cfg.skip_recovery)?,
+        Create => create(wallet_path, cfg.wallet.skip_recovery)?,
         Restore => recover(wallet_path)?,
         Interactive => {
             if path_override {
                 LocalStore::from_file(wallet_path, pwd)?
             } else {
-                interactive(&data_dir)?
+                interactive(&cfg.wallet.data_dir)?
             }
         }
         _ => LocalStore::from_file(wallet_path, pwd)?,
     };
 
     // connect to rusk
-    let rusk = if cfg.ipc_method == "uds" {
-        rusk_uds(cfg.socket_path).await
+    let rusk = if cfg.rusk.ipc_method == "uds" {
+        rusk_uds(&cfg.rusk.socket_path).await
     } else {
-        let rusk_addr = format!("http://{}:{}", cfg.rusk_addr, cfg.rusk_port);
+        let rusk_addr =
+            format!("http://{}:{}", &cfg.rusk.rusk_addr, &cfg.rusk.rusk_port);
```

Comment:
I'm not sure why we are using hardcoded `http` instead of allowing the user to pick `https, etc`

Reply:
This is actually a fair point. Probably for historical reasons but I'd be open to change this too...